openedx/openedx-event-sink-clickhouse
# feat: Add a management command to bulk dump course data to ClickHouse

**Description:** Adds a Django management command for bulk dumping course data to ClickHouse.

**ISSUE:** #5 

**Testing instructions:**

1. In an otherwise working OARS env)
2. Install the OARS plugin branc: https://github.com/openedx/tutor-contrib-oars/pull/35
3. `tutor config save` (to get the ClickHouse config)
4. `tutor images build openedx --no-cache` (to get the plugin installed)
5. `tutor local do init -l oars` (to get the new database and tables created)
6. `tutor local start`
7. Update content in a course and check the "cms worker logs" to see that real-ish time course published sink still functions
8. Shell into the CMS or CMS worker container and experiment with the following commands to see courses dumped to Clickhouse in bulk:

Help: `python manage.py cms dump_courses_to_clickhouse -h`
Dump all eligible courses: `python manage.py cms dump_courses_to_clickhouse`
Dump all courses, no matter their last dump time: `python manage.py cms dump_courses_to_clickhouse --force`
Dump up to 1 course: `python manage.py cms dump_courses_to_clickhouse --limit 1`
Dump to a different ClickHouse than the one in settings: `python manage.py cms dump_courses_to_clickhouse --url https://foo.bar`


You should see updated course data in ClickHouse's event_sink.course_blocks and event_sink.course_relationships


**Reviewers:**
- [ ] @Ian2012 

**Merge checklist:**

**Post merge:**
      finished.
